src/llama-hparams.h

```diff
@@ -115,6 +115,17 @@ struct llama_hparams {
     uint32_t ssm_d_state = 0;
     uint32_t ssm_dt_rank = 0;
     uint32_t ssm_n_group = 0;
+    uint32_t ssm_head_dim   = 0;
+    uint32_t ssm_mamba_d_ssm = 0;
+
+    uint32_t attn_head_dim            = 0;
+    bool     mamba_rms_norm           = false;
+    uint32_t vocab_size               = 0;
```

Comment:
Normally, we don't put the `vocab_size` as `hparam`. Instead, we pick it from the `llama_vocab`. So this is likely not needed.